src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -55,32 +53,39 @@ public GcsPdao(
         this.applicationContext = applicationContext;
     }
 
-    private Storage storageForBucket(GoogleBucketResource bucketResource) {
-        GoogleProjectResource projectResource = bucketResource.getProjectResource();
-        GcsProject gcsProject = gcsProjectFactory.get(projectResource.getGoogleProjectId());
+    public Storage storageForBucket(GoogleBucketResource bucketResource) {
+        GcsProject gcsProject = gcsProjectFactory.get(projectIdForBucket(bucketResource));
         return gcsProject.getStorage();
     }
 
+    public String projectIdForBucket(GoogleBucketResource bucketResource) {
+        return bucketResource.getProjectResource().getGoogleProjectId();
+    }
+
```

Comment:
These also seem like good candidates for utils.

Reply:
They should probably be methods on GoogleBucketResource

Reply:
I fixed the `projectIdForBucket`, i can't easily fix the other one because of the auto wiring.